backend/homework/4/src/main/java/org/q1/StudentUtil.java

```
@@ -0,0 +1,57 @@
+package org.q1;
+public class StudentUtil {
+    public StudentUtil(){
+        
+    }
+    public static double[] calculateGPA(int[] studentIdList, char[][] studentsGrades) throws MissingGradeException {
+        if(studentIdList.length != studentsGrades.length){
+
+            throw new IllegalArgumentException("studentIdList & studentsGrades are out-of-sync. studentIdList.length: "+studentIdList.length + ", studentsGrades.length: " + studentsGrades.length);
+        }
+        double[] gpaList = new double[studentIdList.length];
+        for (int i = 0; i < studentsGrades.length; i++) {
+            double gpa = 0.0;
+            for (int j = 0; j < studentsGrades[i].length; j++) {
+                if (studentsGrades[i][j] == 'A') {
+                    gpa += 4.0;
+                } else if (studentsGrades[i][j] == 'B') {
+                    gpa += 3.0;
+                } else if (studentsGrades[i][j] == 'C') {
+                    gpa += 2.0;
+                } else if (studentsGrades[i][j] == ' ') {
+                    throw new MissingGradeException(studentIdList[i]);
+                }
+            }
+            gpaList[i] = gpa/studentsGrades[i].length;
+        }
+        return gpaList;
+    }
+    public static int[] getStudentsByGPA(double lower, double higher, int[] studentIdList, char[][] studentsGrades) throws InvalidDataException {
+        if (lower < 0 || higher < 0 || lower > higher) {
+            return null;
```

Lets return an exception here